earth_enterprise/src/common/timeutils.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 the Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <time.h>
+
+#include "timeutils.h"
+
+double getMonotonicTime() {
+  struct timespec currTime;
+  clock_gettime(CLOCK_MONOTONIC, &currTime);
+  return static_cast<double>(currTime.tv_sec) + (currTime.tv_nsec * NSEC_TO_SEC);
```

Comment:
Do you think we might run into rounding errors by converting this to a double? Looks like we get about 15 digits safely, give or take. It wouldn't be hard to just store the timespec values until after you do the subtraction. It really depends on what they initialize the clock to. If the number of digits in a time for nanoseconds(9) + seconds(unknown) will exceed 15, then we should do the subtraction using the timespec values. The diff, at least should be within the threshold, but I don't think we can guarantee that an individual instance of a time returned from this function will be.

Reply:
I wish we had C++11.

This is a good idea. I'll make the change.